Ghidra/Processors/6805/data/languages/6x09.sinc

```diff
@@ -0,0 +1,1274 @@
+
+################################################################
+# Opcode 0x00 - 0x0F, relative addressing
+# Opcode 0x40 - 0x4F, register A addressing
+# Opcode 0x50 - 0x5F, register B addressing
+# Opcode 0x60 - 0x6F, indexed addressing
+# Opcode 0x70 - 0x7F, extended addressing
+################################################################
+
+:NEGA    is op=0x40
+{
+        negate(A);
+}
+
+:NEGB 	is op=0x50
+{
+        negate(B);
+}
+
+:NEG OP1    is (op=0x00 | op=0x60 | op=0x70) ... & OP1
+{
+        negate(OP1);
+}
+
+:COMA 	is op=0x43
+{
+        complement(A);
+}
+
+:COMB 	is op=0x53
+{
+        complement(B);
+}
+
+:COM OP1    is (op=0x03 | op=0x63 | op=0x73) ... & OP1
+{
+        complement(OP1);
+}
+
+:LSRA 	is op=0x44
+{
+        logicalShiftRight(A);
+}
+
+:LSRB 	is op=0x54
+{
+        logicalShiftRight(B);
+}
+
+:LSR OP1    is (op=0x04 | op=0x64 | op=0x74) ... & OP1
+{
+        logicalShiftRight(OP1);
+}
+
+:RORA 	is op=0x46
+{
+        rotateRightWithCarry(A);
+}
+
+:RORB 	is op=0x56
+{
+        rotateRightWithCarry(B);
+}
+
+:ROR OP1    is (op=0x06 | op=0x66 | op=0x76) ... & OP1
+{
+        rotateRightWithCarry(OP1);
+}
+
+:ASRA 	is op=0x47
+{
+        arithmeticShiftRight(A);
+}
+
+:ASRB 	is op=0x57
+{
+        arithmeticShiftRight(B);
+}
+
+:ASR OP1    is (op=0x07 | op=0x67 | op=0x77) ... & OP1
+{
+        arithmeticShiftRight(OP1);
+}
+
+:LSLA 	is op=0x48
+{
+        logicalShiftLeft(A);
+}
+
+:LSLB 	is op=0x58
+{
+        logicalShiftLeft(B);
+}
+
+:LSL OP1    is (op=0x08 | op=0x68 | op=0x78) ... & OP1
+{
+        logicalShiftLeft(OP1);
+}
+
+:ROLA    is op=0x49 
+{
+        rotateLeftWithCarry(A);
+}
+ 
+:ROLB    is op=0x59 
+{
+        rotateLeftWithCarry(B);
+}
+ 
+:ROL OP1    is (op=0x09 | op=0x69 | op=0x79) ... & OP1
+{
+        rotateLeftWithCarry(OP1);
+}
+ 
+:DECA    is op=0x4A 
+{
+        decrement(A);
+}
+ 
+:DECB    is op=0x5A 
+{
+        decrement(B);
+}
+ 
+:DEC OP1    is (op=0x0A | op=0x6A | op=0x7A) ... & OP1
+{
+        decrement(OP1);
+}
+ 
+:INCA    is op=0x4C 
+{
+        increment(A);
+}
+ 
+:INCB    is op=0x5C 
+{
+        increment(B);
+}
+ 
+:INC OP1    is (op=0x0C | op=0x6C | op=0x7C) ... & OP1
+{
+        increment(OP1);
+}
+ 
+:TSTA    is op=0x4D 
+{
+        test(A);
+}
+ 
+:TSTB    is op=0x5D 
+{
+        test(B);
+}
+ 
+:TST OP1    is (op=0x0D | op=0x6D | op=0x7D) ... & OP1
+{
+        test(OP1);
+}
+ 
+:JMP OP2    is (op=0x0E | op=0x6E | op=0x7E) ... & OP2
+{
+        goto OP2;
+}
+
+:CLRA    is op=0x4F 
+{
+        clear(A);
+}
+ 
+:CLRB    is op=0x5F 
+{
+        clear(B);
+}
+ 
+:CLR OP1    is (op=0x0F | op=0x6F | op=0x7F) ... & OP1
+{
+        clear(OP1);
+}
+ 
+################################################################
+# Opcode 0x10 - 0x1F, misc. addressing
+################################################################
+
+:NOP    is op=0x12
+{
+}
+
+:SYNC    is op=0x13
+{
+    # TODO wait until not masked interrupt.
```

Comment:
Consider adding pseudo-op here.  I could see reasons not to add it as well, if it appears all over code.
something like WaitNMIClear();

Reply:
Comparing with the 6805: It has a WAIT and STOP instruction. It clears the interrupt mask flag (I) and waits for an external interrupt at the IRQ pin. For this case the ghidra processor simulation clears the I flag but it does not wait for the external interrupt (how to do that within ghidra?).
So I will do it the same way and just remove the TODOs for the SYNC and CWAI instructions.

Reply:
"Prior art" is not necessarily "Prior good practice" ;-)

The real issue is if the effects of the instruction should show up in the decompiler, or if pcode emulation is desired, and the effect is important.

The 6502 should probably have something like:
   I = 0;
   waitInterrupt();
or
   waitInterrupt(I);

I don't think it is actually worth implementing a wait loop on I=1.  Since this is an implicit operation of the processor.  However if it would be important for someone reversing the code to know a wait is done, then it should have a pseudoOp so show up in the decompiler.  It is a model.

In other processors modelers have chosen to do instructions that have no effect on the actual working of the code.

In this case, waiting for an interrupt might be important to the RE.